### DISCUSSION TITLE
Consumer groups for pull based consumers in jetstream?

### DISCUSSION THREAD

**Discussion by achristopoulos-sportserve** [nats-io/nats.deno#705] [Q&A] [answered]
Hello, 
As I have seen in the nats docs (for jetstream), there is the deliver_group for push based consumers however for pull based consumers I do not see any option for consumer groups. For the pull based case, is there any option/alternative to define consumer groups or is this something to be implemented in the future?

**Comment by aricart** [comment, marked as answer]
You don't need consumer groups for pull consumers, just reference the same consumer and fetch/consume/next on it. They will scale horizontally automagically - https://github.com/nats-io/nats.deno/blob/main/jetstream.md#processing-messages

**Comment by achristopoulos-sportserve** [comment]
Thanks for your answer! 
What is the difference between consume and fetch methods?
